volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+                mod_name = PESymbols.os_module_name
+            elif filter_modules_check and not mod_name.endswith(filter_modules_check):
+                continue
+
+            # we won't find symbol information if we can't analyze the module
+            session_layer_name = modules.Modules.find_session_layer(
+                context, session_layers, mod.DllBase
+            )
+            if not session_layer_name:
+                continue
+
+            if mod_name not in found_modules:
+                found_modules[mod_name] = []
+
+            found_modules[mod_name].append(
+                (session_layer_name, mod.DllBase, mod.SizeOfImage)
+            )
+
+        return found_modules
+
+    @staticmethod
+    def get_proc_vads_with_file_paths(
+        proc: interfaces.objects.ObjectInterface,
+    ) -> ranges_type:
+        """
+        Returns a list of the process' vads that map a file
+        """
+        vads = []
+
+        for vad in proc.get_vad_root().traverse():
+            filepath = vad.get_file_name()
+            if not isinstance(filepath, str) or filepath.count("\\") == 0:
+                continue
+
+            vads.append((vad.get_start(), vad.get_size(), filepath))
+
+        return vads
+
+    @classmethod
+    def get_all_vads_with_file_paths(
+        cls,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        symbol_table_name: str,
+    ) -> Generator[
+        Tuple[interfaces.objects.ObjectInterface, str, ranges_type],
+        None,
+        None,
+    ]:
+        """
+        Yields each set of vads for a process that have a file mapped, along with the process itself and its layer
+        """
+        procs = pslist.PsList.list_processes(
+            context=context,
+            layer_name=layer_name,
+            symbol_table=symbol_table_name,
+        )
+
+        for proc in procs:
+            try:
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException:
+                continue
+
+            vads = PESymbols.get_proc_vads_with_file_paths(proc)
+
+            yield proc, proc_layer_name, vads
+
+    @staticmethod
+    def get_process_modules(
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        symbol_table: str,
+        filter_modules: Optional[filter_modules_type],
+    ) -> Dict[str, List[Tuple[str, int, int]]]:
+        """
+        Walks the process list and each process' VAD to determine the base address and size of wanted modules
+        """
+        proc_modules: Dict[str, List[Tuple[str, int, int]]] = {}
+
+        if filter_modules:
+            # create a tuple of module names for use with `endswith`
+            filter_modules_check = tuple([key.lower() for key in filter_modules.keys()])
+        else:
+            filter_modules_check = None
+
+        for _, proc_layer_name, vads in PESymbols.get_all_vads_with_file_paths(
```

Comment:
Would be nice to know what that first underscore would be, any chance you could give it a useful name with an `_` at the start please?  Still a dummy variable, but at least people can more easily refer to a function they might want to use to know what it does.